### PR TITLE
contrib/check-config.sh: fixes for cgroup v2 and kernel v5.x

### LINKED ISSUE #41797: contrib/check-config.sh shows dropped configuration as missing
# `contrib/check-config.sh` uses dropped configurations

Looking at <https://github.com/torvalds/linux/commit/db8ab38880e06dedbfc879e75f5b0ddc495f4eb6>, we see that some configuration parameters were dropped.

**Steps to reproduce the issue:**
1. Run the `contrib/check-config.sh` with linux kernel `>=5.1`
2. See that `CONFIG_NF_NAT_IPV4: missing` and `CONFIG_NF_NAT_NEEDED: missing` is shown

**Describe the results you received:**

The parameters are shown as missing

**Describe the results you expected:**

These parameters are not "missing", they were dropped. This could be misleading.

**Additional environment details (AWS, VirtualBox, physical, etc.):**

`/etc/os-release`

``` BASH
   1   │ NAME="Ubuntu"
   2   │ VERSION="20.04.1 LTS (Focal Fossa)"
   3   │ ID=ubuntu
   4   │ ID_LIKE=debian
   5   │ PRETTY_NAME="Ubuntu 20.04.1 LTS"
   6   │ VERSION_ID="20.04"
   7   │ HOME_URL="https://www.ubuntu.com/"
   8   │ SUPPORT_URL="https://help.ubuntu.com/"
   9   │ BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
  10   │ PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
  11   │ VERSION_CODENAME=focal
  12   │ UBUNTU_CODENAME=focal
```

`$ uname -a`

``` BASH
Linux <name> 5.4.0-58-generic #64-Ubuntu SMP Wed Dec 9 08:16:25 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

Related to https://github.com/docker/docker.github.io/issues/10592

### LINKED ISSUE #41798: contrib/check-config.sh cgroup hierarchy: nonexistent?? with cGroups v2
# `contrib/check-config.sh` shows `cgroup hierarchy: nonexistent??` with cGroups v2

**Steps to reproduce the issue:**
1. Use cGroups v2 on Ubuntu 20.04
2. Use the `contrib/check-config.sh` script to check your ocnfiguration

**Describe the results you received:**

The cgroup hierarchy is shown as nonexistent.

**Describe the results you expected:**

cGroup hierarchy should be working (?)

**Additional information you deem important (e.g. issue happens only occasionally):**

I'm not a professional when it comes to cGroups. Maybe my setup is wrong.

**Additional environment details (AWS, VirtualBox, physical, etc.):**

None I deem important at the moment.

### PR DESCRIPTION
**Pull request by kolyshkin**
fixes https://github.com/moby/moby/issues/41797
fixes https://github.com/moby/moby/issues/41798

1. contrib/check-config.sh: support for cgroupv2
    
Before:
    
> Generally Necessary:
> - cgroup hierarchy: nonexistent??
>     (see https://github.com/tianon/cgroupfs-mount)
    
After:
    
> Generally Necessary:
> - cgroup hierarchy: cgroupv2

2. Make check for various CONFIG_* options recently removed from the kernel conditional.

See individual commits for details.

### CODE REVIEW

File: contrib/check-config.sh [hunk 1]
```diff
@@ -155,18 +155,22 @@ echo
 echo 'Generally Necessary:'
 
 echo -n '- '
-cgroupSubsystemDir="$(awk '/[, ](cpu|cpuacct|cpuset|devices|freezer|memory)[, ]/ && $3 == "cgroup" { print $2 }' /proc/mounts | head -n1)"
-cgroupDir="$(dirname "$cgroupSubsystemDir")"
-if [ -d "$cgroupDir/cpu" ] || [ -d "$cgroupDir/cpuacct" ] || [ -d "$cgroupDir/cpuset" ] || [ -d "$cgroupDir/devices" ] || [ -d "$cgroupDir/freezer" ] || [ -d "$cgroupDir/memory" ]; then
-	echo "$(wrap_good 'cgroup hierarchy' 'properly mounted') [$cgroupDir]"
+if [ "$(stat -f -c %t /sys/fs/cgroup 2>/dev/null)" = "63677270" ]; then
```
Comment (tianon) [inline contrib/check-config.sh:158]:
```suggestion
if [ "$(stat -f -c %t /sys/fs/cgroup 2>/dev/null)" = '63677270' ]; then
```

**Review by tianon — COMMENTED** [review]
:eyes: LGTM

File: contrib/check-config.sh (hunk 1 above)
Comment (thaJeztah) [inline contrib/check-config.sh:158]: Actually looks like shfmt validation is failing on this; it expects a space after `2>`; might indeed as well change to single quotes while at it

```suggestion
if [ "$(stat -f -c %t /sys/fs/cgroup 2> /dev/null)" = '63677270' ]; then
```

**Review by thaJeztah — CHANGES_REQUESTED** [review]
looks good after linting failure is fixed 👍

File: contrib/check-config.sh (hunk 1 above)
Comment (kolyshkin) [reply contrib/check-config.sh:158]: fixed

Comment (kolyshkin) [reply contrib/check-config.sh]: fixed

**Review by tianon — APPROVED** [review]
LGTM

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by AkihiroSuda — APPROVED** [review]
I think we need more validations for cgroup2, but it can be worked out later, so LGTM

**Comment by thaJeztah** [comment]
@AkihiroSuda do you know what's missing? If so, could you open a tracking issue that lists the missing ones?

**Comment by AkihiroSuda** [comment]
opened https://github.com/moby/moby/issues/41874

**Pull request merged**